### ISSUE TITLE
Flavor mapping fails on app center build

### ISSUE THREAD

**Issue by to-dy** [maxkomarychev/react-native-ultimate-config#27]
# Describe the bug

Flavor mapping on appcenter build fails with file not found but works well locally

# Steps to Reproduce

set up flavor mapping according to cook book (yaml file exists in repo)
```
project.ext.flavorEnvMapping = [
    dev: "../.env.yaml",
    staging: "../.env.staging.yaml"
]
```
# Expected behavior

Expected Flavor mapping to be success full on appcenter (works locally)

# Stacktraces, logs, screenshots

```
FAILURE: Build failed with an exception.

* Where:
Script '/Users/runner/runners/2.171.1/work/1/s/node_modules/react-native-ultimate-config/android/rnuc.gradle' line: 17

* What went wrong:
A problem occurred evaluating script.
> /Users/runner/runners/2.171.1/work/1/s/../.env.yaml (No such file or directory)
```

# Environment
macos-10.15
## Integration option
- [x] I am using scheme/flavor based integration (experimental integration
      option from cookbook:
      [scheme](../../docs/cookbook.md#using-multiple-schemes-ios),
      [flavors](../../docs/cookbook.md#using-multiple-flavors-android))

## OS

- [x] android

## Packages

- version of `react-native`: 0.62.2
- version of `react-native-ultimate-config`: ^3.1.4

# Additional context

build works fine locally

**Comment by maxkomarychev** [comment]
@to-dy hi, can you please share more information on how exactly do you do build in appcenter? what command do you use? from which folder do you win it?


thanks.

**Comment by to-dy** [comment]
@maxkomarychev 
it's a[ Gradle task ](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/gradle?view=azure-devops)
command can be found below
```
[command]/Users/runner/runners/2.171.1/work/1/s/android/gradlew -DAPPCENTER_KEYSTORE_PASSWORD=*** -DMOBILECENTER_KEYSTORE_PASSWORD=*** -DAPPCENTER_KEY_ALIAS=*** -DMOBILECENTER_KEY_ALIAS=*** -DAPPCENTER_KEY_PASSWORD=*** -DMOBILECENTER_KEY_PASSWORD=*** -DAPPCENTER_BUILD_VERSION=184 -DMOBILECENTER_BUILD_VERSION=184 -p android clean :app:assembleDevRelease :app:bundleDevRelease
Downloading https://services.gradle.org/distributions/gradle-6.1.1-all.zip
...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
Unzipping /Users/runner/.gradle/wrapper/dists/gradle-6.1.1-all/cfmwm155h49vnt3hynmlrsdst/gradle-6.1.1-all.zip to /Users/runner/.gradle/wrapper/dists/gradle-6.1.1-all/cfmwm155h49vnt3hynmlrsdst
Set executable permissions for: /Users/runner/.gradle/wrapper/dists/gradle-6.1.1-all/cfmwm155h49vnt3hynmlrsdst/gradle-6.1.1/bin/gradle

Welcome to Gradle 6.1.1!

Here are the highlights of this release:
 - Reusable dependency cache
 - Configurable compilation order between Groovy/Kotlin/Java/Scala
 - New sample projects in Gradle's documentation

For more details see https://docs.gradle.org/6.1.1/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)
Gradle 6.1.1

> Configure project :
Resolved com.android.tools.build:gradle:4.0.0 in :classpath 
Resolved com.google.gms:google-services:4.3.3 in :classpath 
Project myCareAI at : is either no Android app project or build version has not been set to override. Skipping...

> Configure project :app
Resolved org.yaml:snakeyaml:1.19 in classpath 
Flavor mapping detected: [dev:../.env.yaml, staging:../.env.staging.yaml]

FAILURE: Build failed with an exception.

* Where:
Script '/Users/runner/runners/2.171.1/work/1/s/node_modules/react-native-ultimate-config/android/rnuc.gradle' line: 17

* What went wrong:
A problem occurred evaluating script.
> /Users/runner/runners/2.171.1/work/1/s/../.env.yaml (No such file or directory)
```

**Comment by maxkomarychev** [comment]
would it be possible to see full piece of config relevant for this command? or  any chance you are specifying custom `cwd` parameter there?

it looks like relative path is computed against root of the project instead of android folder.
I don't have experience with appcenter so for now I can only speculate.


could you try changing your config to be

```
project.ext.flavorEnvMapping = [
    dev: ".env.yaml",
    staging: ".env.staging.yaml"
]
```

it might break the local build but I am wondering if this will do the trick for appcenter

meanwhile I'll try to think of more robust way of specifying path to config file.

**Comment by maxkomarychev** [comment]
ok I found the problem. gradle process is launched from root of the repository while existing code expects current working directory to be "android" folder. I am sure there is a simple fix to this issue.

**Comment by to-dy** [comment]
@maxkomarychev thanks for your time on this. 🙏🏾

**Comment by maxkomarychev** [comment]
@to-dy I have potential fix in branch "android-relative-paths" (pr https://github.com/maxkomarychev/react-native-ultimate-config/pull/28). I'll merge and publish it as soon as e2e is ok, you can meanwhile try installing it directly from the branch to give it a try.

thanks

**Comment by maxkomarychev** [comment]
@to-dy I have published new version [3.1.5](https://github.com/maxkomarychev/react-native-ultimate-config/blob/master/CHANGELOG.md#315-2020-06-29) which should fix this problem. Please feel free to reopen if issue occurs again.

thanks for feedback.
